lib/sles4sap/console_redirection.pm

```diff
@@ -102,65 +201,70 @@ This allows OpenQA access to command return codes and output for evaulation by s
 
 sub connect_target_to_serial {
     my (%args) = @_;
-    $args{'target_ip'} //= get_required_var('REDIRECT_TARGET_IP');
-    $args{'ssh_user'} //= get_required_var('REDIRECT_TARGET_USER');
+    $args{target_ip} //= get_required_var('REDIRECT_TARGET_IP');
+    $args{ssh_user} //= get_required_var('REDIRECT_TARGET_USER');
 
-    croak "OpenQA variable WORKER_VM_ID undefined. Run 'redirection_init()' first" unless get_var('WORKER_VM_ID');
-    croak "IP address '$args{'target_ip'}' is not valid." unless grep(/^$RE{net}{IPv4}$/, $args{'target_ip'});
+    # Save original value for 'AUTOINST_URL_HOSTNAME', and point requests to localhost
+    # check os-autoinst/testapi.pm host_ip() function to get an idea about inner workings
```

Comment:
this is an hack that introduce dependency to an inner API of testapi. Isn't it? What if they decide to change the name of `AUTOINST_URL_HOSTNAME` ?

Reply:
It's a global backend variable. It's there for this exact reason. So you can change stuff in your test.

Reply:
https://github.com/os-autoinst/os-autoinst/blob/master/doc/backend_vars.asciidoc